crates/provider/src/fillers/nonce.rs

```diff
@@ -40,11 +114,18 @@ use std::sync::Arc;
 /// # }
 /// ```
 #[derive(Clone, Debug, Default)]
-pub struct NonceFiller {
-    nonces: DashMap<Address, Arc<Mutex<u64>>>,
+pub struct NonceFiller<M: NonceManager> {
```

Comment:
could we make the CachedNonceMg the default type here? so that we can rid of type hints

Reply:
ok, I'll try but why CachedNonceManager and not SimpleNonceManager? see PR description about reorgs

Reply:
I don't have a preference, also fine with SimpleNonceManager as the default, good point